lib/tls/tls.cc

```diff
@@ -275,7 +231,7 @@ namespace
 		size_t     readyLength;
 		Capability readyBuffer =
 		  br_ssl_engine_sendrec_buf(engine, &readyLength);
-		precisely_bound_buffer(readyBuffer, readyLength);
+		readyBuffer.bounds().set_inexact_at_most(readyLength);
```

Comment:
I think line 241 now needs to be .length of this? Otherwise, I would expect a bounds failure.

Reply:
`set_inexact_at_most` takes its argument by reference and updates it, so the length of `readBuffer`'s bounds should equal `readyLength`?  Am I missing something?

Reply:
Oh, hmm, I missed that. It is inconsistent with the other CHERI::Capability APIs, so we should probably change it or rename the API to make it explicit.

Reply:
Sorry, I just stole the API from here.  Open to suggestions for amendments.

I guess when we do switch to `CSetBoundsRoundDown`, we'll have to follow it with the `CGetLength` that this write-back elides.  Still, probably pairing those two in this function is a fine thing to do.

Reply:
I would rather remove the writeback. The get length is an explicit operation and the use is easier to follow. I *think* the compiler can elide an unused bounds-get intrinsic, but I’m not 100% sure.